lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -902,19 +903,40 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
     configuration = configuration.copyWith(locales: parseBrowserLanguages());
   }
 
-  static List<ui.Locale> parseBrowserLanguages() {
-    // TODO(yjbanov): find a solution for IE
-    final List<String>? languages = domWindow.navigator.languages;
+  // The country codes that use Hant (chinese traditional) script
+  static const Set<String> _hantCountryCodes = <String>{'TW', 'HK', 'MO', 'CHT'};
+
+  /// Converts a list of strings representing languages [domWindow.navigator.languages] into [ui.Locale].
+  ///
+  /// For tests, the list of languages can be overridden using the [debugLanguages] parameter.
+  static List<ui.Locale> parseBrowserLanguages({
+    @visibleForTesting List<String>? debugLanguages,
+  }) {
+    final List<String>? languages = debugLanguages ?? domWindow.navigator.languages;
     if (languages == null || languages.isEmpty) {
       // To make it easier for the app code, let's not leave the locales list
       // empty. This way there's fewer corner cases for apps to handle.
       return const <ui.Locale>[_defaultLocale];
     }
 
     final List<ui.Locale> locales = <ui.Locale>[];
+
+    // TODO(dit): Use Intl.Locale to parse each language string, instead of doing it by hand.
```

Comment:
Can you please file an issue for this?

Reply:
Yes!

Reply:
https://github.com/flutter/flutter/issues/130174 Let me link it in the PR while the tree is red